.nx/workflows/agents.yaml

```diff
@@ -0,0 +1,64 @@
+parallelism: 8
+env:
+  CI: 'true'
+  GIT_AUTHOR_EMAIL: test@test.com
+  GIT_AUTHOR_NAME: Test
+  GIT_COMMITTER_EMAIL: test@test.com
+  GIT_COMMITTER_NAME: Test
+  NX_E2E_CI_CACHE_KEY: e2e-circleci-linux
+  NX_VERBOSE_LOGGING: 'true'
+  NX_DAEMON: 'true'
+  NX_PERF_LOGGING: 'false'
+  NX_NATIVE_LOGGING: 'false'
+  SELECTED_PM: 'pnpm'
+  NX_E2E_RUN_E2E: 'true'
+  NPM_CONFIG_PREFIX: '/home/workflows/.npm-global'
+steps:
+  - name: Git Clone
+    script: |
+      git init .
+      git remote add origin $GIT_REPOSITORY_URL
+      git fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +{{nxCommitSha}}:{{nxCommitRef}}
+      git checkout --progress --force -B {{nxBranch}} {{nxCommitRef}}
+
+  - name: Restore cache
+    script: |
+      nxw cache restore {{nxBranch}}-node_modules node_modules
```

Comment:
How does this work between the commits? Will every commit on the same branch use the same cached artifact despite potential dependency changes?